File: JavaScript/7js 2/generator.js
/*
|--------------------------------------------------------------------------
| Multiplication - Generators #2
|--------------------------------------------------------------------------
|
*/

function* generator(a) {
   for (let x = 1; ; x++) yield `${a} x ${x} = ${a * x}`;
}


function* generator(a, b = 1) {
   while (true) yield `${a} x ${b} = ${a * b++}`;
}


function* generator(a) {
   let b = 0;
   while (++b) {
      yield `${a} x ${b} = ${a * b}`;
   }
}

File: Python/7py 2/generator.py
"""
|--------------------------------------------------------------------------
| Multiplication - Generators #2
|--------------------------------------------------------------------------
|
"""

def generator(a):
   x = 1
   while True:
      yield f"{a} x {x} = {a * x}"
      x += 1


generator = lambda a: (f"{a} x {i} = {a * i}" for i in __import__("itertools").count(1))


def generator(a):
   return (f'{a} x {i} = {a * i}' for i in range(1,11))